### PR TITLE
fix: off by one in db refcounting

### PR DESCRIPTION
**Pull request by matklad**
This feels like a typo to me, but I have next to zero understanding of this code, so I would prefer that someone else takes over this, if this is a bug at all.

### CODE REVIEW

**Review by bowenwang1996 — COMMENTED** [review]
I believe this is intentional. @mikhailOK can explain more

**Comment by matklad** [comment]
I suspected that! then, the comment above the function might need an update?

**Comment by mikhailOK** [comment]
It's sort of impossible case. `rc == 0` is serialized as empty:
```
/// Refcount record format:
/// rc = 0 => empty
/// rc < 0 => 8 bytes little endian rc
/// rc > 0 => value followed by 8 bytes little endian rc
```

`merge_refcounted_records` and `encode_value_with_rc` have this invariant, so before that line of code we can have `debug_assert!(rc != 0)`.

Also, `rc < 0` is possible in individual records we insert, but never in the output of `merge_refcounted_records` because rocksdb merges in the order of insertion.

**Comment by matklad** [comment]
Makes sense! Changed the commit to fix the comment instead!

File: core/store/src/db/refcount.rs [hunk 1]
```diff
@@ -50,8 +50,8 @@ pub(crate) fn merge_refcounted_records(result: &mut Vec<u8>, val: &[u8]) {
 }
 
 /// Returns
-/// (Some(value), rc) if rc > 0
-/// (None, rc) if rc <= 0
+/// (Some(value), rc) if rc >= 0
+/// (None, rc) if rc < 0
```
Comment (bowenwang1996) [inline core/store/src/db/refcount.rs:54]: but if rc == 0 the value doesn't exist, so the orginal comment seems correct?

Comment (matklad) [reply core/store/src/db/refcount.rs:54]: Hmmmm, right, we then hit the first branch, and return `(None, 0)`, that makes sesne!

**Pull request closed without merging**